### PR TITLE
feat: metrics generation workflow

### PR DESCRIPTION
**Pull request by SachinSahu431**
This PR adds a bash script that will generate new metric files whenever a new Go version is released. After that, a PR will be raised, containing all these files.

Closes: https://github.com/prometheus/client_golang/issues/1435

### CODE REVIEW

File: .github/workflows/update-go-versions.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---
+name: Weekly Go Version Check and Metric Generation
+
+on:
+  schedule:
+    - cron: "0 0 * * 0"  # Every Sunday at midnight UTC
```
Comment (ArthurSens) [inline .github/workflows/update-go-versions.yml:6]: According to [go release docs](https://go.dev/wiki/Go-Release-Cycle), new releases are created in February and August. Maybe we could limit this workflow to run only during those months?

**Review by ArthurSens — COMMENTED** [review]
Have you tried this workflow on a fork?

**Comment by SachinSahu431** [comment]
> Have you tried this workflow on a fork?

I tried it on a seperate fork, but was unable to trigger it manually. I added `workflow_dispatch` to do do, but this workflow was not visible in Actions tab. Am I missing something?

**Comment by SachinSahu431** [comment]
The bash script will generate new metric files whenever a new Go version is detected. Then a PR will be raised, containing all these files. 

Example PR: https://github.com/SachinSahu431/client_golang/pull/2

**Comment by SachinSahu431** [comment]
Hey!

Here is one example of autogenerated PR for Go version 1.22: https://github.com/SachinSahu431/client_golang/pull/2

Please let me know if we should modify the format of PR. For example: adding some labels, marking it draft by default, etc.

**Comment by SachinSahu431** [comment]
Note that in [supported_go_versions.txt](https://github.com/SachinSahu431/client_golang/pull/2/files#diff-c99d42eb6b3d2c81f2a74179e4c0909b956fc104da0f19499d28e0129f2be58c), older Go versions are not being removed. Out of all versions listed in this txt file, only the top 3 versions are being used by [update-go-version.bash file](https://github.com/prometheus/client_golang/pull/1481/files#diff-e2c042a423598946ade46fd28f670b306073b19b82cf1b2bcd1ca09050bdc390) even if 4th (and later) versions are supported.

File: update-go-version.bash [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# Check for new version of Go, and generate go collector test files
+# New Go version gets appended at top of supported_go_versions.txt, and only top 3 versions are supported
+if [[ ! $current_version =~ $latest_version ]]; then
+  echo "New Go version available: $latest_version"
+  echo "Updating supported_go_versions.txt and generating Go Collector test files"
+  sed -i "1i $latest_version" supported_go_versions.txt
+  make generate-go-collector-test-files
+else
+  echo "No new Go version detected. Current Go version is: $current_version"
+fi
```
Comment (kakkoyun) [inline update-go-version.bash:21]:
```suggestion
fi

```

File: .github/workflows/update-go-versions.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+---
+name: Generate Metric files for new Go version
+
+on:
+  workflow_dispatch:
+  schedule:
+    # Runs weekly during month of February and August
+    # Reference: https://go.dev/wiki/Go-Release-Cycle
+    - cron: '0 0 * 2,8 1'
```
Comment (kakkoyun) [inline .github/workflows/update-go-versions.yml:9]: The document doesn't state the actual day of the release so maybe we should run it on the last day of the month.

Still, this wouldn't be very fail-safe; what happens if the release is stalled or postponed? I don't see any downsides to running the task every month. It's a no-op if there isn't any version released.

Comment (SachinSahu431) [reply .github/workflows/update-go-versions.yml:9]: Agreed 👍
Let's schedule it to run on a monthly basis.

**Comment by ArthurSens** [comment]
> Note that in [supported_go_versions.txt](https://github.com/SachinSahu431/client_golang/pull/2/files#diff-c99d42eb6b3d2c81f2a74179e4c0909b956fc104da0f19499d28e0129f2be58c), older Go versions are not being removed. Out of all versions listed in this txt file, only the top 3 versions are being used by [update-go-version.bash file](https://github.com/prometheus/client_golang/pull/1481/files#diff-e2c042a423598946ade46fd28f670b306073b19b82cf1b2bcd1ca09050bdc390) even if 4th (and later) versions are supported.

I think this will cause confusion in the future 😬. Could we make sure that we only have the supported versions in the `supported_go_versions` file? I.e. the 3 latest go versions

**Comment by SachinSahu431** [comment]
@ArthurSens made the changes, thanks! 🙌

**Review by kakkoyun — APPROVED** [review]

**Pull request merged**